store/types/utils.go

```diff
@@ -105,3 +106,12 @@ func PrefixEndBytes(prefix []byte) []byte {
 func InclusiveEndBytes(inclusiveBytes []byte) []byte {
 	return append(inclusiveBytes, byte(0x00))
 }
+
+// Marshaler defines a interface needed for the store package to marshal data
+type Marshaler interface {
+	// Marshal returns binary encoding of v.
+	Marshal(proto.Message) ([]byte, error)
```

Comment:
is there any reason we should use the codec here? Im not sure how it provides value in the store package when we are encoding a proto type? @aaronc do you happen to know?

Reply:
Probably safer to use something like this than totally remove it - in case there's a need. This is kind of like an expected keeper for `Codec` which makes sense.

Reply:
so it doesn't seem like this is possible unless I refactor the binarycodec as well since in app.go it needs a different argument. Do you know a way to avoid this

Reply:
The codec interface is wrong. I remember bringing this up before. It should take `proto.Message` and not `ProtoMarshaler`